packages/xrpl/src/sugar/submit.ts

```diff
@@ -148,7 +148,7 @@ async function waitForFinalTransactionOutcome(
     .catch(async (error) => {
       // error is of an unknown type and hence we assert type to extract the value we need.
       // eslint-disable-next-line @typescript-eslint/consistent-type-assertions,@typescript-eslint/no-unsafe-member-access -- ^
-      const message = error.data.error as string
+      const message = error?.data?.error as string
```

Comment:
Can you add a test for this?

Reply:
I'm not really sure how to - in order to test this I'd need to cause a `tx` request to error without any text in the error case, and I'm not sure when that happens. My guess is that this change will mostly re-skin the problem so the actual error that's arising can now be caught when it happens.